nncf/checkpoint_loading.py

```diff
@@ -38,97 +41,154 @@ def load_state(model: torch.nn.Module, state_dict_to_load: dict, is_resume: bool
     :return: The number of state_dict_to_load entries successfully matched and loaded into model.
     """
 
-    def key_normalizer(key):
-        new_key = key
-        match = re.search('(pre_ops|post_ops)\\.(\\d+?)\\.op', key)
-        return new_key if not match else new_key.replace(match.group(), 'operation')
-
     if 'state_dict' in state_dict_to_load:
         state_dict_to_load = state_dict_to_load['state_dict']
     model_state_dict = model.state_dict()
 
-    new_dict, num_loaded_layers, problematic_keys = match_keys(is_resume, state_dict_to_load, model_state_dict,
-                                                               key_normalizer)
-    num_saved_layers = len(state_dict_to_load.items())
-    process_problematic_keys(is_resume, problematic_keys, num_loaded_layers == num_saved_layers)
+    key_matcher = KeyMatcher(is_resume, state_dict_to_load, model_state_dict)
+    new_dict, num_loaded_layers = key_matcher.run()
+    key_matcher.handle_problematic_keys()
     nncf_logger.info("Loaded {}/{} layers".format(num_loaded_layers, len(model_state_dict.items())))
 
     model.load_state_dict(new_dict, strict=False)
     return num_loaded_layers
 
 
-def process_problematic_keys(is_resume, issues, is_all_saved_loaded):
-    error_msgs = []
+class OptionalParametersRegistry:
```

Comment:
Do any of the current parameters in our checkpoints fall into the category of "optional"?

Reply:
not now, but I think it makes sense to add 'enabled' and '_num_bits' of Quantizer. I will add them.

If there're any other candidates of the optional parameters (e.g. in pruning? @mkaglins, @AlexanderDokuchaev), please let me know or feel free to register throw `OptionalParametersRegistry`